Sources/NIOWebSocket/Base64.swift

```diff
@@ -128,3 +121,51 @@ internal struct Base64 {
     return alphabet[Int(index)]
   }
 }
+
+extension String {
+  /// This is a backport of a proposed String initializer that will allow writing directly into an uninitialized String's backing memory.
+  ///
+  /// As this API does not exist prior to 5.3 on Linux, or on older Apple platforms, we fake it out with a pointer and accept the extra copy.
+  @inlinable
+  init(backportUnsafeUninitializedCapacity capacity: Int,
+       initializingUTF8With initializer: (_ buffer: UnsafeMutableBufferPointer<UInt8>) throws -> Int) rethrows {
+    // The buffer will store zero terminated C string
+    let buffer = UnsafeMutableBufferPointer<UInt8>.allocate(capacity: capacity + 1)
+    defer {
+        buffer.deallocate()
+    }
+
+    let initializedCount = try initializer(buffer)
+    precondition(initializedCount <= capacity, "Overran buffer in initializer!")
+    // add zero termination
+    buffer[capacity] = 0
```

Comment:
We need to null-terminate at `initializedCount`, not at `capacity`, or we risk using uninitialised memory here.